src/renderer/components/layout/topbar.tsx

```diff
@@ -22,16 +22,44 @@
 import styles from "./topbar.module.css";
 import React from "react";
 import { observer } from "mobx-react";
+import { TopBarRegistry } from "../../../extensions/registries";
 
 interface Props extends React.HTMLAttributes<any> {
   label: React.ReactNode;
 }
 
 export const TopBar = observer(({ label, children, ...rest }: Props) => {
+  const renderRegisteredItems = () => {
+    const items = TopBarRegistry.getInstance().getItems();
+
+    if (!Array.isArray(items)) {
+      return null;
+    }
+
+    return (
+      <div className="px-6">
+        {items.map((registration, index) => {
+          if (!registration?.components?.Item) {
+            return null;
+          }
+
+          return (
+            <div key={index}>
+              {registration.components.Item}
+            </div>
+          );
+        })}
```

Comment:
nit: filter might be easier to read.

```suggestion
        {
          items
            .filter(registration => registration?.components?.Item)
            .map(({ components: { Item } }, index) => (
              <div key={index}>
                {Item}
              </div>
            ))
        }
```

Also, shouldn't the props be passed in somewhere?